src/main/java/io/cdap/plugin/servicenow/source/ServiceNowBaseSourceConfig.java

```diff
@@ -163,39 +163,36 @@ private void validateValueType(FailureCollector collector) {
 
   private void validateDateRange(FailureCollector collector) {
     if (containsMacro(ServiceNowConstants.PROPERTY_START_DATE) ||
-      containsMacro(ServiceNowConstants.PROPERTY_END_DATE)) {
+      containsMacro(ServiceNowConstants.PROPERTY_END_DATE) ||
+      (Util.isNullOrEmpty(startDate) && Util.isNullOrEmpty(endDate))) {
       return;
     }
 
-    if (!Util.isNullOrEmpty(startDate) && !Util.isNullOrEmpty(endDate) &&
-      !Util.isValidDateFormat(ServiceNowConstants.DATE_FORMAT, startDate) &&
-      !Util.isValidDateFormat(ServiceNowConstants.DATE_FORMAT, endDate)) {
-      collector.addFailure("Invalid format for Start date. Correct Format: " +
-          ServiceNowConstants.DATE_FORMAT, null)
+    if (Util.isNullOrEmpty(startDate) && !Util.isNullOrEmpty(endDate)) {
+      collector.addFailure("Start date is empty.", null)
         .withConfigProperty(ServiceNowConstants.PROPERTY_START_DATE);
-      collector.addFailure("Invalid format for End date. Correct Format:" +
-          ServiceNowConstants.DATE_FORMAT, null)
+      return;
+    }
+
+    if (Util.isNullOrEmpty(endDate) && !Util.isNullOrEmpty(startDate)) {
+      collector.addFailure("End date is empty.", null)
```

Comment:
collector.addFailure("An end date must be given when a start date is given.", "Enter an end date, or remove the start date")

Reply:
Update the failure message